File: app/src/main/java/org/kindone/willingtodo/persistence/sqlite/ConfigSqliteHelper.java
package org.kindone.willingtodo.persistence.sqlite;

import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteStatement;
import android.util.Log;

import org.kindone.willingtodo.data.TaskContext;

/**
 * Created by kindone on 2016. 12. 14..
 */

public class ConfigSqliteHelper {
    final static String TAG = "ConfigSqliteHelper";
    final String tableName;

    public ConfigSqliteHelper(String tableName) {
        this.tableName = tableName;
    }

    public void createTableIfNotExists(SQLiteDatabase db) {
        db.execSQL("CREATE TABLE " + tableName + "(_id INTEGER PRIMARY KEY AUTOINCREMENT," + " key TEXT NOT NULL, value TEXT NOT NULL)");
        db.execSQL("CREATE UNIQUE INDEX " + tableName + "_KEY_IDX ON " + tableName + "(key)");
    }

    public int getCurrentTabIndex(SQLiteDatabase db) {
        return getIntValue(db, "tab_index", 0);
    }

    public void setCurrentTabIndex(SQLiteDatabase db, int index) {
        setRawValue(db, "tab_index", String.valueOf(index));
    }

    private ContentValues prepareContentValues(String key, String value) {
        ContentValues values = new ContentValues();
        values.put("key", key);
        values.put("value", value);

        return values;
    }

    private String getRawValue(SQLiteDatabase db, String key) {
        Cursor cursor = db.rawQuery("select value from " + tableName + " where key = ?", new String[]{key});
        if(cursor.moveToNext()) {
            return cursor.getString(0);
        }
        return null;
    }

    private int getIntValue(SQLiteDatabase db, String key, int defaultValue) {
        Cursor cursor = db.rawQuery("select value from " + tableName + " where key = ?", new String[]{key});
        if(cursor.moveToNext()) {
            String value = cursor.getString(0);
            return Integer.valueOf(value);
        }
        return defaultValue;
    }


    private int setRawValue(SQLiteDatabase db, String key, String value) {
        Log.v(TAG, "Set " + key + " = " + value);
        SQLiteStatement stmt = db.compileStatement("INSERT OR REPLACE INTO " + tableName + "(key,value) values(?, ?)");
        stmt.bindAllArgsAsStrings(new String[]{key, value});
        int numRowsAffected = stmt.executeUpdateDelete();
        return numRowsAffected;
    }

}

File: app/src/main/java/org/kindone/willingtodo/persistence/sqlite/ConfigSqliteHelper.kt
package org.kindone.willingtodo.persistence.sqlite

import android.content.ContentValues
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteStatement
import android.util.Log

import org.kindone.willingtodo.data.TaskContext

/**
 * Created by kindone on 2016. 12. 14..
 */

class ConfigSqliteHelper(internal val tableName: String) {

    fun createTableIfNotExists(db: SQLiteDatabase) {
        db.execSQL("CREATE TABLE $tableName(_id INTEGER PRIMARY KEY AUTOINCREMENT, key TEXT NOT NULL, value TEXT NOT NULL)")
        db.execSQL("CREATE UNIQUE INDEX " + tableName + "_KEY_IDX ON " + tableName + "(key)")
    }

    fun getCurrentTabIndex(db: SQLiteDatabase): Int {
        return getIntValue(db, "tab_index", 0)
    }

    fun setCurrentTabIndex(db: SQLiteDatabase, index: Int) {
        setRawValue(db, "tab_index", index.toString())
    }

    private fun prepareContentValues(key: String, value: String): ContentValues {
        val values = ContentValues()
        values.put("key", key)
        values.put("value", value)

        return values
    }

    private fun getRawValue(db: SQLiteDatabase, key: String): String? {
        val cursor = db.rawQuery("select value from $tableName where key = ?", arrayOf(key))
        if (cursor.moveToNext()) {
            return cursor.getString(0)
        }
        return null
    }

    private fun getIntValue(db: SQLiteDatabase, key: String, defaultValue: Int): Int {
        val cursor = db.rawQuery("select value from $tableName where key = ?", arrayOf(key))
        if (cursor.moveToNext()) {
            val value = cursor.getString(0)
            return Integer.valueOf(value)!!
        }
        return defaultValue
    }


    private fun setRawValue(db: SQLiteDatabase, key: String, value: String): Int {
        Log.v(TAG, "Set $key = $value")
        val stmt = db.compileStatement("INSERT OR REPLACE INTO $tableName(key,value) values(?, ?)")
        stmt.bindAllArgsAsStrings(arrayOf(key, value))
        val numRowsAffected = stmt.executeUpdateDelete()
        return numRowsAffected
    }

    companion object {
        internal val TAG = "ConfigSqliteHelper"
    }

}
